File: encoders/qls/src/main/java/nl/uva/sc/encoders/qls/ast/Widget.java
package nl.uva.sc.encoders.qls.ast;

import nl.uva.sc.encoders.ql.ast.TextLocation;

public class Widget extends AstNode {

	private final String name;

	public Widget(TextLocation textLocation, String name) {
		super(textLocation);
		this.name = name;
	}

	public String getWidgetName() {
		return name;
	}
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/StyleAttributes/Widgets/Widget.cs
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model.StyleAttributes.Widgets
{
    abstract public class Widget : StyleAttribute
    {
        protected Widget(TextPosition position)
            : base(position) { }

        abstract public bool SupportsDataType(DataType dataType);
    }
}
